src/NUnitEngine/Addins/nunit.v2.driver/XmlExtensions.cs

```diff
@@ -96,8 +96,8 @@ private static XmlNode AddTest(this XmlNode parent, ITest test, bool recursive)
             if (test.IsSuite)
                 thisNode.AddAttribute("type", test.TestType);
             thisNode.AddAttribute("id", test.TestName.TestID.ToString());
-            thisNode.AddAttribute("name", Escape(test.TestName.Name));
```

Comment:
As commented on the issue, we have to create valid XML. We escape the name and fullname because they may contain arbitrary characters inside string arguments as well as for generic fixtures, which use angle brackets.

I'm pretty sure this will break the XML output. Note that we don't need to do this for id, runstate, etc. because we know we don't use any special characters in those fields, but name and fullname come from the user.